core/src/main/java/org/infinispan/xsite/irac/DefaultIracManager.java

```diff
@@ -219,10 +222,12 @@ public void onTopologyUpdate(CacheTopology oldCacheTopology, CacheTopology newCa
       }
 
       if (addedSegments.isEmpty()) {
-         // this node doesn't have any new segments but it may become the primary owner
-         // of some.
-         // trigger a new round (it also removes the keys if the node isn't a owner)
-         iracExecutor.run();
+         // this node doesn't have any new segments, but it may become the primary owner of some segments.
+         // i.e. backup owner => primary owner "promotion"
+         if (!updatedKeys.isEmpty()) {
```

Comment:
The comment should be inside the `if`.

But I don't really get the purpose of the `if`, since `run()` is very lightweight when `updatedKeys.isEmpty()`. The only difference is with this `if` you skip the `hasClear` check.

Reply:
which comment? 

`hasClear` is only set to `true` in the originator and it does not check ownership of the segments. 

Yes `run()` is lightweight but I got tired to see a thread spawning and logs about it when the server starts.

Reply:
I meant the comment on lines 225-226, in this position it's unclear to me if it refers to the if on line 224, to the if on line 227, or to the iracManager.run() call on line 228. I was assuming the latter.

IMO it would be better if `run()` checked if `updatedKeys.isEmpty()` itself before logging anything.

Reply:
`iracExecutor.run()` does not know anything about what is being run.
Checking the map in `org.infinispan.xsite.irac.DefaultIracManager#run()` is not useful for me because, at this point, a new thread is started

Reply:
I see. I thought the annoyance was only the trace log in `DefaultIracManager#run()`, because IracExecutor uses the blocking executor, and I highly doubt the server can start without spawning at least one blocking thread.

Reply:
Looking at this method again, wouldn't it be better if the if on line 226 was inverted?
That way you wouldn't have the duplicated `updatedKeys.isEmpty()` that bothers me so much :)

```
      if (!addedSegments.isEmpty()) {
         // group new segments by primary owner
         Map<Address, IntSet> primarySegments = new HashMap<>();
         for (int segment : addedSegments) {
            Address primary = newCacheTopology.getWriteConsistentHash().locatePrimaryOwnerForSegment(segment);
            primarySegments.computeIfAbsent(primary, DefaultIracManager::newIntSet).add(segment);
         }

         primarySegments.forEach(this::sendStateRequest);
      }

      // Even if this node doesn't have any new segments, it may become the primary owner of some segments.
      // i.e. backup owner => primary owner "promotion"
      // only trigger a round if we have pending updates.
      if (!updatedKeys.isEmpty()) {
         iracExecutor.run();
      }
```

Reply:
sure